x/tx/signing/context.go

```diff
@@ -14,6 +14,8 @@ import (
 	"cosmossdk.io/core/address"
 )
 
+const maxRecursionDepth = 32
```

Comment:
Do you think there is a use case to make this as a field in the context options that default to 32, to be future proof?
That allows us to add a test for the max recursion as well.